### DISCUSSION TITLE
Handling cache with authentication

### DISCUSSION THREAD

**Discussion by bacqueyrisses** [vercel/next.js#67634] [App Router] [answered]
Hey,

I'm using AuthJS with Next.js 14, and I'm encountering an issue with Server Components opting out of caching when calling the auth() function ([due to cookies](https://nextjs.org/docs/app/building-your-application/caching#invalidation-1), it seems).

I tried moving the auth() function into the fetch function itself, but I'm still experiencing the same behavior.

Is there any way to cache the session? How do you handle this issue?

Thanks! 

![CleanShot 2024-07-10 at 18 52 29@2x](https://github.com/vercel/next.js/assets/96829831/c8a4b808-451c-4e83-9556-fb9a19a157e4)
![CleanShot 2024-07-10 at 18 40 52@2x](https://github.com/vercel/next.js/assets/96829831/c7dc4397-ad96-4a8e-a67d-d883afd9d075)

**Comment by versecafe** [comment]
You could use react's `cache()` to cache all the auth calls in a given request
https://react.dev/reference/react/cache

**Reply by bacqueyrisses** [reply]
I already tried it, but the cache still misses. I might have implemented it incorrectly.

![CleanShot 2024-07-10 at 19 31 31@2x](https://github.com/vercel/next.js/assets/96829831/51f98b88-478a-4b42-b84a-b669c903697e)
![CleanShot 2024-07-10 at 19 31 43@2x](https://github.com/vercel/next.js/assets/96829831/41836e2e-0f50-47d8-a327-1a90b609e3fb)

**Reply by versecafe** [reply]
the react cache should apply if you have multiple `authCached` calls being made during a single network request, are you trying to cache across multiple?

**Reply by bacqueyrisses** [reply]
Oh I see. I don't, so it will be useful for the next session function call. Thank you for clarifying

**Comment by bacqueyrisses** [comment, marked as answer]
Resolved: Using[ unstable_cache](https://nextjs.org/docs/app/api-reference/functions/unstable_cache) solves this issue.
